### DISCUSSION TITLE
Get Platform and Architecture at Build Time

### DISCUSSION THREAD

**Discussion by seanlaw** [pypa/cibuildwheel#910] [Q&A]
Hi all,

I am using Github Actions and `cibuildwheel` to build wheels for Windows `x64` and `win32`. Additionally, I have a `CIBW_BEFORE_BUILD: bash ./build.sh` script that executes right before building the wheel but contains a condition (if/else) that depends on whether it is an `x64` build (i.e., omit a `cmake` flag) OR a `win32` build (i.e., add a `cmake` flag). I was wondering if there was any way to add some logic inside of the `build.sh` script to detect that `cibuildwheel` is currently building a wheel for `win32` so that I can set my `cmake` flag accordingly?

So, inside of `build.sh`, I'd be looking to do something like:

```
if [ "$PLATFORM" == "Windows" && "$ARCH" == "win32"]; then
    # Set cmake flag here
fi
```

Any help would be greatly appreciated!

**Comment by louisabraham** [comment]
Why not use `CIBW_ARCHS` to restrict the architecture and use two different jobs (one for x86 and one for x64)?

**Reply by seanlaw** [reply]
Definitely possible. I was hoping for something a bit more straightforward like the `if` condition :). It will just make the Github Actions workflow much longer than it already is.

**Comment by henryiii** [comment]
Could we set an extra environment variable or two, perhaps, to make this a bit easier? We could set CIBW_CURRENT_SELECTOR to the current selector inside the (before) build steps, and maybe CIBW_ARCH (no S) inside build-all and before-build?

Splitting the build by CIBW_ARCH likely is nice for doing parallel builds, so not a bad solution, but just thinking of a possible general solution.

**Reply by seanlaw** [reply]
Yes, I think that would be helpful! And I agree that a general solution is best. Essentially, to some extent, I think my condition/question is satisfied as long as I can "introspect" (??) from within a given step (i.e., before build) the current architecture and platform (along with any other `cibuildwheel` condition) that is unique to the current build environment.

**Reply by joerick** [reply]
This seems pretty reasonable to me. In terms of naming, we already set `CIBUILDWHEEL=1` for our builds, so perhaps we could add `CIBUILDWHEEL_BUILD_IDENTIFIER` - the `CIBUILDWHEEL` prefix distinguishing these variables from `CIBW` options.

**Reply by hynek** [reply]
I would _love_ to have something like `CIBW_CURRENT_SELECTOR` / `CIBW_ARCH`.

In argon2-cffi-bindings I need to distinguish whether I'm cross-compiling and my [current approach](https://github.com/hynek/argon2-cffi-bindings/pull/2/files#diff-f6251d5dde136ce99df0d00b95dd5fd910782fd0df8f7dba99d3f91002354515l) feels super janky and I expect it to break in weird ways in the most unfortunate moment.

**Reply by joerick** [reply]
Thanks for the feedback! Created an issue for this feature request #944
